libs/config/common/src/lib/endpoints.ts

```diff
@@ -479,24 +476,26 @@ export class Endpoints {
       )}`;
     } else {
       return `${storageEndpoint}/${this.getSimulationRunS3Path(
-        runId,
-      )}/contents/${fileLocation}`;
+        runId, `contents/${fileLocation}`
+        )}`;
     }
   }
+
   /**
    * Create a path a simulation run in an S3 bucket
    * @param runId Id of the simulation run
    */
-  public getSimulationRunS3Path(runId: string): string {
-    return `${this.simulationRunsS3Path}/${runId}`;
+  public getSimulationRunS3Path(runId: string, subPath?: string): string {
+    subPath = subPath ? `/${subPath}` : '';
+    return `${this.simulationRunsS3Path}/${runId}${subPath}`;
   }
 
   /**
    * Create a path for the COMBINE/OMEX archive of a simulation run in an S3 bucket
    * @param runId Id of the simulation run
    */
   public getSimulationRunCombineArchiveS3Path(runId: string): string {
-    return `${this.getSimulationRunS3Path(runId)}/archive.omex`;
+    return this.getSimulationRunS3Path(runId, 'archive.omex');
```

Comment:
Rather than rely on endpoints class to encode the paths here, I think it would be best to have the details/knowledge of the file organization on s3 all contained within the simulation-storage service and remove the dependency on endpoints entirely. Nothing else in the system needs to know how the s3 buckets are organized once the combine archives are extracted since they all then have URLs attached to them and are saved in the database. Keeping this here just encourages some other part of the code to rely on this information, causing coupling where it does not need to be

Reply:
Feel free to create a separate issue to move this.

Reply:
The configuration for the S3 paths is shared between the frontend and backend. This has to be separated before this configuration can be organized into the shared storage library.

Reply:
yes, this is the main problem. The front end really shouldn't be aware of the s3 bucket at all, but get any files through the API.  
I will take a look at this when I return next week

Reply:
I separated this and organized the S3 paths into the shared storage library.